samples/ar-view-node/src/main/java/io/github/sceneview/sample/arviewnode/nodes/TrackingNode.kt

```diff
@@ -0,0 +1,341 @@
+package io.github.sceneview.sample.arviewnode.nodes
+
+import android.content.Context
+import android.util.Log
+import android.view.View
+import android.widget.ImageView
+import androidx.core.content.ContextCompat
+import com.google.ar.core.DepthPoint
+import com.google.ar.core.Frame
+import com.google.ar.core.HitResult
+import com.google.ar.core.Plane
+import com.google.ar.core.Point
+import com.google.ar.core.Trackable
+import com.google.ar.core.TrackingState
+import com.google.ar.core.exceptions.DeadlineExceededException
+import com.google.ar.core.exceptions.FatalException
+import com.google.ar.core.exceptions.NotTrackingException
+import com.google.ar.sceneform.rendering.ViewAttachmentManager
+import com.google.ar.sceneform.rendering.ViewRenderable
+import dev.romainguy.kotlin.math.Float3
+import dev.romainguy.kotlin.math.Quaternion
+import io.github.sceneview.ar.ARSceneView
+import io.github.sceneview.ar.arcore.createAnchorOrNull
+import io.github.sceneview.ar.arcore.position
+import io.github.sceneview.ar.node.AnchorNode
+import io.github.sceneview.math.toRotation
+import io.github.sceneview.node.ViewNode
+import io.github.sceneview.sample.arviewnode.FrameTime
+import io.github.sceneview.sample.arviewnode.MainActivity
+import io.github.sceneview.sample.arviewnode.R
+
+
+class TrackingNode(
+    private val context: Context,
+    private val sceneView: ARSceneView
+) {
+
+    /**
+     * Indicates if the Camera is currently tracking.
+     */
+    private var isTracking = false
+
+    /**
+     * Indicates if the HitTest was successful.
+     */
+    private var isHitting = false
+
+    /**
+     * If we skip Frames this value is set to true, otherwise false.
+     */
+    private var isHittestPaused = true
+
+    private var createdCalled = false
+
+    /**
+     * This is a simple counter to track skipped frames. After HITTEST_SKIP_AMOUNT is
+     * reached this counter is set back to 0.
+     */
+    private var currentHittestAttempt = 0
+
+    /**
+     * That is our HitResult. This variable is the most important one in this
+     * class. Based on the HitResult Anchors can be created.
+     */
+    private var lastHitResult: HitResult? = null
+
+    /**
+     * This is the timestamp of the used Frame to perform the HitTest.
+     */
+    private var lastHitTimestamp: Long = 0
+
+    private var lastFps = 0.0
+
+    private var frameTime: FrameTime? = null
+
+    private var isValid = ValidIndicator.IS_NOT_VALID
+
+    private var trackingAnchorNode: AnchorNode? = null
```

Comment:
I'm not sure if we need an anchor node for visualizing the latest hit point. However, the current implementation shouldn't cause any problems.

Reply:
You are right, it is not necessary. Fixed it.